### DISCUSSION TITLE
Next.JS project failed to deploy "Type error: Layout "app/(platform)/(dashboard)/board/[boardId]/layout.tsx" does not match the required types of a Next.js Layout. "generateMetaData" is not a valid Layout export field."

### DISCUSSION THREAD

**Discussion by youssefash97** [community/community#123840] [Programming Help]
### Body

I tried to deploy a NextJs project and it works fine when i tested it but when i tried to deploy it on Vercel it gave me this error in image
![vercel error](https://github.com/community/community/assets/126786341/46b034f7-fab7-46a5-b57c-9dc6fd1f1ff1)

My code:

`import { db } from "@/lib/db";
import { auth } from "@clerk/nextjs";
import { notFound, redirect } from "next/navigation";
import { BoardNavbar } from "./_components/board-navbar";

export async function generateMetaData({
  params,
}: {
  params: { boardId: string };
}) {
  const { orgId } = auth();
  if (!orgId) {
    return {
      title: "Board",
    };
  }
  const board = await db.board.findUnique({
    where: {
      id: params.boardId,
      orgId,
    },
  });

  return {
    title: board?.title || "Board",
  };
}

const BoardIdLayout = async ({
  children,
  params,
}: {
  children: React.ReactNode;
  params: { boardId: string };
}) => {
  const { orgId } = auth();

  if (!orgId) {
    redirect("/select-org");
  }

  const board = await db.board.findUnique({
    where: {
      id: params.boardId,
      orgId,
    },
  });

  if (!board) {
    notFound();
  }

  return (
    <div
      className="relative h-full bg-no-repeat bg-cover"
      style={{ backgroundImage: `url(${board.imageFullUrl})` }}
    >
      <BoardNavbar data={board} />
      <div className="absolute inset-0 bg-black/10" />
      <main className="relative pt-28 h-full">{children}</main>
    </div>
  );
};

export default BoardIdLayout;
`

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by Ahmadalhomsi** [comment]
I suggest you to read the dynamic routes in NextJs documentation.
[https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes](https://nextjs.org/docs/pages/building-your-application/routing/dynamic-routes)
